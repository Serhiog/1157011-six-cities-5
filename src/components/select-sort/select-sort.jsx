import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../store/action";


const SelectSort = ({handleSorting}) => {
  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span className="places__sorting-type" tabIndex={0}>
        Popular
        <svg className="places__sorting-arrow" width={7} height={4}>
          <use xlinkHref="#icon-arrow-select" />
        </svg>
      </span>
      <select onChange={handleSorting} className="places__sorting-type" id="places-sorting">
        <option className="places__option" value="popular" defaultValue="">
          Popular
        </option>
        <option className="places__option" value="to-high">
          Price: low to high
        </option>
        <option className="places__option" value="to-low">
          Price: high to low
        </option>
        <option className="places__option" value="top-rated">
          Top rated first
        </option>
      </select>
    </form>
  );
};

SelectSort.propTypes = {
  handleSorting: PropTypes.func,
};

const mapToStateProps = (state) => ({
  unicCities: state.offers.unicCities,
});

const mapDispatchToProps = (dispatch) => ({
  handleSorting(evt) {
    dispatch(ActionCreator.handleSorting(evt.target.value));
  },
});

export {SelectSort};
export default connect(mapToStateProps, mapDispatchToProps)(SelectSort);
