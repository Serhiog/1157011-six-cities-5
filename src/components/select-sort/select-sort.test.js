import React from "react";
import renderer from "react-test-renderer";
import {SelectSort} from "../select-sort/select-sort";

it(`Should Offer render correctly`, () => {
  const tree = renderer
    .create(<SelectSort handleSorting={() => {}} />, {
      createNodeMock: () => {
        return {};
      },
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
