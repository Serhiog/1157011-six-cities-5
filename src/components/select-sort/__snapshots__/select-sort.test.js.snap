// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Offer render correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
    style={
      Object {
        "marginRight": "10px",
      }
    }
  >
    Sort by
  </span>
  <select
    className="places__sorting-type"
    id="places-sorting"
    onChange={[Function]}
  >
    <option
      className="places__option"
      defaultValue=""
      value="popular"
    >
      Popular
    </option>
    <option
      className="places__option"
      value="to-high"
    >
      Price: low to high
    </option>
    <option
      className="places__option"
      value="to-low"
    >
      Price: high to low
    </option>
    <option
      className="places__option"
      value="top-rated"
    >
      Top rated first
    </option>
  </select>
</form>
`;
